addons/crm/crm_lead_menu.xml

```diff
@@ -38,7 +39,8 @@
         <field name="context">{
                 'stage_type': 'opportunity',
                 'default_type': 'opportunity',
-                'default_user_id': uid
+                'default_user_id': uid,
+                'toolbar_hide_actions': ['crm.action_mark_as_archive'],
```

Comment:
no need to set this option for opportunity action.. because there is not archive action in opportunity list view..